controller/search_blackbox_test.go

```diff
@@ -2000,3 +2001,48 @@ func TestWorkItemInterfaceSliceSort(t *testing.T) {
 		require.Equal(t, WorkItemInterfaceSlice{a, b, c}, s)
 	})
 }
+
+func (s *searchControllerTestSuite) TestSearchByTrackerQuery() {
+	fxt := tf.NewTestFixture(s.T(), s.DB,
+		tf.Spaces(1),
+		tf.WorkItemTypes(1),
+		tf.Trackers(1),
+		tf.TrackerQueries(2),
+		tf.WorkItems(5, func(fxt *tf.TestFixture, idx int) error {
+			switch idx {
+			case 0, 1, 2, 3:
+				fxt.WorkItems[idx].Fields[workitem.SystemRemoteTrackerID] = fxt.TrackerQueries[0].ID
+			default:
+				fxt.WorkItems[idx].Fields[workitem.SystemRemoteTrackerID] = fxt.TrackerQueries[1].ID
+			}
+			return nil
+		}),
+	)
+	assert.NotNil(s.T(), fxt.Spaces, fxt.Trackers, fxt.WorkItemTypes, fxt.TrackerQueries, fxt.WorkItems)
+	spaceIDStr := fxt.Spaces[0].ID.String()
+
+	s.T().Run("search by trackerquery", func(t *testing.T) {
```

Comment:
@DhritiShikhar you need to test for a non-existing tracker query ID in this test as well and ensure that no error occurs.

Reply:
That's right. I missed that part. I have created an issue for the same -> https://github.com/fabric8-services/fabric8-wit/issues/2396

I will raise a PR for it.